src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1265,6 +1295,111 @@ static void fill_idPtypeNP
   const std::string&  typePatternQ
 )
 {
+  /*
+   * Given a subscription document, for each item within the entities array field, the condition is true if:
+   *
+   * - the item isPattern field is "true" AND
+   * - the item isTypePattern field is false OR doesn't exist AND
+   * - the item type is empty string OR doesn't exist OR is equal to the target type (entityType) AND
+   * - the item id matches the target id (entityId)
+   *
+   * Example of query block generated by this function:
+   *
+   * {
+   *   "entities.isPattern": "true",
+   *   "entities.isTypePattern": {
+   *     "$ne": true
+   *   },
+   *   "expiration": {
+   *     "$gt": 1666606572
+   *   },
+   *   "status": {
+   *     "$ne": "inactive"
+   *   },
+   *   "$expr": {
+   *     "$anyElementTrue": {
+   *       "$map": {
+   *         "input": "$entities",
+   *         "in": {
+   *           "$and": [
+   *             {
+   *               "$regexMatch": {
+   *                 "input": "(entityId)",
+   *                 "regex": "$$this.id",
+   *                 "options": "i"
+   *               }
+   *             },
+   *             {
+   *               "$eq": [
+   *                 "$$this.isPattern",
+   *                 "true"
+   *               ]
+   *             },
+   *             {
+   *               "$or": [
+   *                 {
+   *                   "$eq": [
+   *                     "$$this.type",
+   *                     "(entityType)"
+   *                   ]
+   *                 },
+   *                 {
+   *                   "$eq": [
+   *                     "$$this.type",
+   *                     ""
+   *                   ]
+   *                 },
+   *                 {
+   *                   "$eq": [
+   *                     {
+   *                       "$type": "$$this.type"
+   *                     },
+   *                     "missing"
+   *                   ]
+   *                 }
+   *               ]
+   *             },
+   *             {
+   *               "$or": [
+   *                 {
+   *                   "$eq": [
+   *                     "$$this.isTypePattern",
+   *                     false
+   *                   ]
+   *                 },
```

Comment:
This block is probably redundant, given the:

```
   "entities.isTypePattern": {
     "$ne": true
   },
```

Reply:
Fixed in 3653139

In this case, I have left the part in $expr as I think is clearer